Patch-Source: https://github.com/getting-things-gnome/gtg/commit/2ecc997d3fb3f6d102615dafa83bda765cebd420
From 2ecc997d3fb3f6d102615dafa83bda765cebd420 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz93@gmail.com>
Date: Sun, 3 Dec 2023 17:41:54 -0500
Subject: [PATCH] migrate from external mock package to stdlib unittest.mock

It was moved to the stdlib in python 3.3, and the PyPI version is a
backport for python versions less than 3.3.
---
 .github/workflows/unit_tests.yml      | 4 ++--
 README.md                             | 4 ++--
 flatpak/python3-requirements-dev.yaml | 9 ---------
 flatpak/requirements-dev.txt          | 1 -
 tests/backend/backend_caldav_test.py  | 2 +-
 tests/core/test_config.py             | 2 +-
 tests/core/test_timer.py              | 2 +-
 tests/test_utils.py                   | 2 +-
 8 files changed, 8 insertions(+), 18 deletions(-)

diff --git a/tests/backend/backend_caldav_test.py b/tests/backend/backend_caldav_test.py
index 521d7ae9c1..51d594a5d0 100644
--- a/tests/backend/backend_caldav_test.py
+++ b/tests/backend/backend_caldav_test.py
@@ -11,7 +11,7 @@
 from GTG.core.datastore import DataStore
 from GTG.core.dates import LOCAL_TIMEZONE, Date
 from GTG.core.task import Task
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 from tests.test_utils import MockTimer
 
 NAMESPACE = 'unittest'
diff --git a/tests/core/test_config.py b/tests/core/test_config.py
index 6eee15e9b7..2607a9134d 100644
--- a/tests/core/test_config.py
+++ b/tests/core/test_config.py
@@ -19,7 +19,7 @@
 from unittest import TestCase
 import configparser
 
-from mock import patch, mock_open, Mock
+from unittest.mock import patch, mock_open, Mock
 
 from GTG.core.config import open_config_file, SectionConfig
 
diff --git a/tests/core/test_timer.py b/tests/core/test_timer.py
index ada5074451..fb7d5841bd 100644
--- a/tests/core/test_timer.py
+++ b/tests/core/test_timer.py
@@ -18,7 +18,7 @@
 
 import datetime
 
-from mock import patch
+from unittest.mock import patch
 from unittest import TestCase
 
 from GTG.core.timer import Timer
diff --git a/tests/test_utils.py b/tests/test_utils.py
index e9903e136a..6db524b514 100644
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -1,4 +1,4 @@
-from mock import Mock
+from unittest.mock import Mock
 
 
 class MockThread(Mock):
